feat: [CDS-80530]: Document rollback behaviour for traditional deployments (#5425)
harness/developer-hub: docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng.md

* `<+instance.host.privateIp>`
* `<+instance.host.publicIp>`

### Rollback

> Harness will restore back the state of deployment to previous pipeline successfull stage execution based on `serivce`, `enviroment` and `infrastucture` details.

Reviewer: ```suggestion
Harness restores the state of deployment to the pipeline's previous successful stage execution based on `serivce`, `enviroment` and `infrastucture` details.
```

> Harness will keep a record on the artifact version that was successfully deployed during previous successfull executions - This would mean that if the rollback step contains Copy Artifact command unit, Harness will copy the last successfull version of the artifact (Deployed via Harness) to remote host.

Reviewer: ```suggestion
Harness records the artifact version that was successfully deployed during previous successful executions. When using the Rollback step's Copy Artifact command unit, Harness copies the last successful version of the artifact deployed via Harness to the remote host.
```

#### First time deployment

> In case if first pipeline execution fails (regardless of stage) - Harness will skip the rollback since there is no record of any successfull pipeline execution.

Reviewer: ```suggestion
If the first pipeline execution fails (regardless of stage), Harness skips the rollback since there is no record of any successful pipeline execution.
```

#### N+1 time deployment

> In case of stage failures in subsequent executions ((Assume the service, environment, infrastructure didn't changed in corresponding stage)) - Harness will initiate the rollback to the previous successfull pipeline with stage execution. Previous pipeline execution needs not to be successfull for all stages, it will match the successfull stage regardless of the pipeline execution status.

Reviewer: ```suggestion
In case of stage failures in subsequent executions ((Assuming the service, environment, infrastructure didn't changed in the corresponding stage)), Harness initiates rollback to the previous successful pipeline. The previous pipeline execution must be successful for all stages.  The successful stage is matched regardless of the pipeline execution status.
```

:::note

> * If any of the `service`, `environemnt`, `infrastructure` details were changed in stage, Harness won't consider any previous successfull pipeline executions for rollback, this is treated as completely different deployment.

Reviewer: ```suggestion
* If any of the `service`, `environment`, or `infrastructure` details were changed in the stage, Harness won't consider any previous successful pipeline executions for rollback. It is treated as a completely different deployment.
```

> * In case of multiple stages referencing same `service`, `environment` and `infrastructure` details, Harness will rollback the deployemnt to last successfull pipeline with stage execution that shared same `service`, `environment` and `infrastructure`.

Reviewer: ```suggestion
* In case of multiple stages referencing the same `service`, `environment`, and `infrastructure` details, Harness will rollback the deployment to last successful pipeline with stage execution that shared the same `service`, `environment`, and `infrastructure`.
```
